### DISCUSSION TITLE
A Field View Template

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#6803] [Q&A] [answered]
In TW 5.2.2 and older version, I used a template like below to display fields in view mode

```
<$vars newFieldNameTiddler=<<qualify "$:/temp/NewFieldName">>
newFieldValueTiddler=<<qualify "$:/temp/NewFieldValue">>  
searchListState=<<qualify "$:/temp/searchListState">> 
storeTitle=<<qualify "$:/temp/storeTitle">> 
>
<$transclude tiddler="$:/core/ui/EditTemplate/fields"/>
</$vars>
```

In TW 5.2.3p, this does not work, so I followed to see what has been changed and released  https://github.com/Jermolene/TiddlyWiki5/pull/6628 has been merged. It adds a feature to keep the input value when the filed name is changed. I updated my template to reflect this

```
\import [[$:/core/ui/EditTemplate]]
<!-- the above import required from TW5.2.3-->
<$reveal type=match stateTiddler stateTitle="$:/state/utility/view-fields" stateField=text default="hide" text=show tag=div class="kk-ut-viewfields">
<$vars newFieldNameTiddler=<<qualify "$:/temp/NewFieldName">>
newFieldValueTiddler=<<qualify "$:/temp/NewFieldValue">>  
newFieldNameInputTiddler=<<qualify "$:/temp/NewFieldName/input">>
newFieldNameSelectionTiddler=<<qualify "$:/temp/NewFieldName/selected-item">>
searchListState=<<qualify "$:/temp/searchListState">> 
storeTitle=<<qualify "$:/temp/storeTitle">> 
>
<$transclude tiddler="$:/core/ui/EditTemplate/fields"/>
</$vars>
</$reveal>
```

To give a try, open https://tiddlywiki.com/prerelease/, create a tiddler with above code and tag with `$:/tags/ViewTemplate`
Then create a new tiddler, save and try add a field. TW UI crashes.

I am wondering what else happened to `$:/core/ui/EditTemplate/fields` that the solution does not work anymore.

Any idea?

**Comment by kookma** [comment, marked as answer]
I understood the new fields section is more complicated than what is in TW 5.2.2. It seems some variables are added which is not present in my update, so the solution breaks TW UI.

For any one interested, for TW5.2.3 the below template works

```
<$reveal type=match stateTiddler stateTitle="$:/state/utility/view-fields" stateField=text default="hide" text=show tag=div class="kk-ut-viewfields">
<$importvariables filter="[[$:/core/ui/EditTemplate]]">
<!-- the above import required from TW5.2.3-->
<$vars 
	newFieldNameTiddler=<<qualify "$:/temp/NewFieldName">>
	newFieldValueTiddlerPrefix=<<qualify "$:/temp/NewFieldValue">>
	newFieldNameInputTiddler=<<qualify "$:/temp/NewFieldName/input">>
	newFieldNameSelectionTiddler=<<qualify "$:/temp/NewFieldName/selected-item">>

newFieldValueTiddler=<<qualify "$:/temp/NewFieldValue">> 

searchListState=<<qualify "$:/temp/searchListState">> 
storeTitle=<<qualify "$:/temp/storeTitle">> 
>
<$transclude tiddler="$:/core/ui/EditTemplate/fields"/>
</$vars>
</$importvariables>
</$reveal>

<!-- note that the
newFieldValueTiddler=<<qualify "$:/temp/NewFieldValue">> 
is kept for backward compatibility. TW 5.2.2 and older
-->
```

**Comment by Jermolene** [comment]
Hi @kookma that's disappointing. As you know, we try to retain rigorous backwards compatibility. #6511 has given us a few problems – #6628 is actually to fix a change in UI behaviour that was introduced in #6511.

We could fix the issue you raise here by ensuring that `$:/core/ui/EditTemplate/fields` applies default values for missing parameters. I'll convert it to an issue.

**Reply by kookma** [reply]
Thank you @Jermolene.
As I explained above, I could revise my solution and it works in TW 5.2.3.
I use a button on page control in [Utility plugin ](https://kookma.github.io/TW-Utility) to reveals fields in view mode.
I will push a new update to support TW 5.2.3. While the old release can work with TW 5.2.2 and older.
So, for me it is fine to keep the new changes in TW 5.2.3

Just a side note:
To my understanding the core codes get more complicated, it is good to be able to simply follow the code and be able to debug!

**Comment by kookma** [comment]
There was report on misbehavior of fields in edit mode when upgrade a TW 5.2.2 to latest TW 5.2.3
https://talk.tiddlywiki.org/t/field-height-position-not-ok-after-upgrade-to-v-5-2-3-release/4162/4
